engine/runtime/src/main/java/org/enso/interpreter/node/controlflow/caseexpr/StringLiteralBranchNode.java

```diff
@@ -39,9 +42,35 @@ void doText(
     }
   }
 
+  @Specialization
+  void doJavaString(VirtualFrame frame, Object state, String target) {
+    if (textProfile.profile(equalStrings(literal, target))) {
+      accept(frame, state, new Object[0]);
+    }
+  }
```

Comment:
I think this one is spurious/redundant but wait on @JaroslavTulach 's opinion

Reply:
Yes, all _string like_ objects should be wrapped into `Text` in Enso. E.g. the bug/fix should really be in `HostValueToEnsoNode`

Reply:
So I should remove the `doJavaString` branch, but after all I _should keep_ the `doInteropString` branch, as the `HostValueToEnsoNode` is already doing everything it can do - and we just have to keep in mind that we must not only accept `Text` but also `isString` interop objects.

Is my understanding correct @JaroslavTulach ?

Reply:
> HostValueToEnsoNode is already doing everything it can do

It does, when it gets called. Which it apparently didn't until [these changes](https://github.com/enso-org/enso/pull/9203#issuecomment-1968896105).

> we must not only accept `Text` but also `isString` interop objects.

Probably we should rewrite `ToJavaStringNode` to accept not only `Text`, but _also `isString` interop objects_.

> should keep the `doInteropString` branch,

Yes, very likely we should keep it.

Reply:
> Probably we should rewrite `ToJavaStringNode` to accept not only `Text`, but _also `isString` interop objects_.

But `ToJavaStringNode` is described as `"Converts Enso Text to a Java String."`.

We probably will still keep separate execution paths for `Text` type and `isString`, so I'm not sure if we should be modifying `ToJavaStringNode`. At least it does not seem needed for this PR, if I understand things correctly.